app/models/account.rb

```diff
@@ -48,9 +48,9 @@ def update_level
                  .map { |l| [l.key, l.value].join ':' }
     levels = Level.all.order(id: :asc)
 
-    levels.each do |lvl|
+    levels.each_with_index do |lvl, index|
       break unless tags.include?(lvl.key + ':' + lvl.value)
-      account_level = lvl.id
+      account_level = index + 1
```

Comment:
to me lvl.id is the level, and your change is using the array index
why would you like to use the array index ?

Reply:
to be able to edit levels in future:
index is actually an order id of levels database, so this way, for example, you will be able to delete level 3(if you suddenly need to) and what was level 4 will become level 3(by order, not by id), or you will be able even to reinitiolize all the levels by deleting them all and creating new ones